client/src/Molecules.elm

```diff
@@ -226,46 +389,180 @@ level1Decoder =
             )
             Nothing
         |> JDP.required "responsible" Decode.string
+        |> JDP.required "group" Decode.string
         |> JDP.optional "notes" (Decode.maybe Decode.string) Nothing
-        |> JDP.required "sequence" Decode.string
+        |> JDP.required "sequence_length" Decode.int
         |> JDP.required "children" (Decode.list level0Decoder)
-        -- TODO: This should be properly decoded
+        |> JDP.optional "backbone" (Decode.maybe backboneDecoder) Nothing
         |> JDP.hardcoded Nothing
 
 
+
+-- TODO: This should be properly decoded
+
+
+dispatchDecoder : Decode.Decoder Vector
+dispatchDecoder =
+    Decode.field "level" Decode.int
+        |> Decode.andThen
+            (\level ->
+                case level of
+                    1 ->
+                        Decode.andThen (Decode.succeed << BackboneVec) backboneDecoder
+
+                    2 ->
+                        Decode.andThen (Decode.succeed << Level0Vec) level0Decoder
+
+                    3 ->
+                        Decode.andThen (Decode.succeed << LevelNVec) level1Decoder
+
+                    _ ->
+                        Decode.fail "Unknow Level!"
+            )
+
+
+vectorDecoder : Decode.Decoder (List Vector)
+vectorDecoder =
+    Decode.list dispatchDecoder
+
+
+
+-- Encoders
+
+
+backboneEncoder : Backbone -> Encode.Value
+backboneEncoder backbone =
+    Encode.object
+        [ ( "name", Encode.string backbone.name )
+        , ( "location", Encode.int backbone.location )
+        , ( "bsa1_overhang"
+          , Encode.string <|
+                showBsa1Overhang <|
+                    Maybe.withDefault InvalidBsa1 backbone.bsa1Overhang
+          )
+        , ( "bsmb1_overhang"
+          , Encode.string <|
+                showBsmb1Overhang <|
+                    Maybe.withDefault InvalidBsmb1 backbone.bsmb1Overhang
+          )
+        , ( "sequence", Encode.string "NNNNNNNNNNNNNNNNN---NNNNNNN" ) -- TODO: We do need the sequence!!!
```

Comment:
This should come from a genbank file no?

Reply:
Yes indeed! This is an older comment.
It will be deleted in the next commit.